In [4]:
import numpy as np
import pandas as pd

In [6]:
df = pd.read_csv('market_basket_analysis_raw_data.csv', sep=';')

C:\Users\jer93\AppData\Local\Temp\ipykernel_3552\2928347773.py:1: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('market_basket_analysis_raw_data.csv', sep=';')


In [7]:
df.head()

BillNo                             Itemname  Quantity              Date  \
0  536365   WHITE HANGING HEART T-LIGHT HOLDER         6  01.12.2010 08:26   
1  536365                  WHITE METAL LANTERN         6  01.12.2010 08:26   
2  536365       CREAM CUPID HEARTS COAT HANGER         8  01.12.2010 08:26   
3  536365  KNITTED UNION FLAG HOT WATER BOTTLE         6  01.12.2010 08:26   
4  536365       RED WOOLLY HOTTIE WHITE HEART.         6  01.12.2010 08:26   

  Price  CustomerID         Country  
0  2,55     17850.0  United Kingdom  
1  3,39     17850.0  United Kingdom  
2  2,75     17850.0  United Kingdom  
3  3,39     17850.0  United Kingdom  
4  3,39     17850.0  United Kingdom

# Data Dictionary

## Fields Description

| Field Name  | Description                                      | Data Type |
|-------------|--------------------------------------------------|-----------|
| `BillNo`    | 6-digit number assigned to each transaction.     | Nominal   |
| `Itemname`  | Product name.                                    | Nominal   |
| `Quantity`  | The quantities of each product per transaction.  | Numeric   |
| `Date`      | The day and time when each transaction was generated. | Numeric   |
| `Price`     | Product price.                                   | Numeric   |
| `CustomerID`| 5-digit number assigned to each customer.        | Nominal   |
| `Country`   | Name of the country where each customer resides. | Nominal   |


In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 522064 entries, 0 to 522063
Data columns (total 7 columns):
 #   Column      Non-Null Count   Dtype  
---  ------      --------------   -----  
 0   BillNo      522064 non-null  object 
 1   Itemname    520609 non-null  object 
 2   Quantity    522064 non-null  int64  
 3   Date        522064 non-null  object 
 4   Price       522064 non-null  object 
 5   CustomerID  388023 non-null  float64
 6   Country     522064 non-null  object 
dtypes: float64(1), int64(1), object(5)
memory usage: 27.9+ MB


## Dataset Overview from `.info()` Call

- **Total Entries:** The dataset consists of 522,064 entries.
- **Columns:** There are 7 columns in the dataset.
- **Non-Null Counts:**
  - `BillNo`, `Quantity`, `Date`, `Price`, and `Country` have no missing values (522,064 non-null).
  - `Itemname` has some missing values (520,609 non-null).
  - `CustomerID` has a significant number of missing values (388,023 non-null).
- **Data Types:**
  - `BillNo`, `Itemname`, `Date`, `Price`, and `Country` are of type `object`, which often indicates categorical or textual data.
  - `Quantity` is an integer (`int64`), suitable for numerical operations.
  - `CustomerID` is a floating point number (`float64`), which is interesting as IDs are typically integers. This warrants further investigation.
- **Memory Usage:** The dataset consumes approximately 27.9 MB of memory.

### Initial Observations and Actions
- **Missing Data:** The missing values in `Itemname` and `CustomerID` need attention. Strategies for handling these missing values should be considered.
- **Data Types:** The data types of `BillNo`, `Date`, `Price`, and `CustomerID` might need conversion depending on their usage. For instance, `Price` could be converted to a numerical type for calculations. `Date` should be converted to datetime. 
- **Further Exploration:** Detailed exploration is required for understanding the distribution of values in each column, especially for `CustomerID` and `Itemname`.


In [11]:
# Assuming df is your DataFrame
df['Date'] = pd.to_datetime(df['Date'], format='%d.%m.%Y %H:%M')
df['Price'] = df['Price'].str.replace(',', '.').astype(float)

In [12]:
df.describe()

Quantity                           Date          Price  \
count  522064.000000                         522064  522064.000000   
mean       10.090435  2011-07-04 12:51:20.777107456       3.826801   
min     -9600.000000            2010-12-01 08:26:00  -11062.060000   
25%         1.000000            2011-03-28 10:15:00       1.250000   
50%         3.000000            2011-07-20 08:59:00       2.080000   
75%        10.000000            2011-10-19 14:12:00       4.130000   
max     80995.000000            2011-12-09 12:50:00   13541.330000   
std       161.110525                            NaN      41.900599   

          CustomerID  
count  388023.000000  
mean    15316.931710  
min     12346.000000  
25%     13950.000000  
50%     15265.000000  
75%     16837.000000  
max     18287.000000  
std      1721.846964